### PR TITLE
A dedicated "events" section on Cassandra website

### PR DESCRIPTION
**Pull request by hemidactylus**
This is a proposed "Events" section on the Cassandra website, listing (approved) events as small cards with a picture, a short description and a link to the external event website.

The motivation behind this idea, which was discussed briefly on the cassandra-website Slack channel as well, is twofold: on one hand, it is desirable to provide this information to website visitors, which helps building the community through dedicated events; on the other hand, the Cassandra blog is preferrably not 'clogged' with too many event announcements so that it can retain its primary "technical" role.

I tried to create a page that blends in with the rest of the website experience, but while doing so I had to make some choices. In particular, I propose an icon for the section (a circus tent) that I hope conveys a "playful but not silly" message; moreover, I added the "Events" page to the "Community" navbar menu to avoid adding another menu, even though all other "Community" menu items are from the community page.
Also I am aware that the buttons alignment on the cards is imperfect. I blame my (very limited) css knowledge, this is of course something to improve if ever this draft gets accepted.

events1

events2

### CODE REVIEW

**Comment by michaelsembwever** [comment]
**Everyone** is intending this to (and knows that it must be) be fair and open to all that wish to organise and sponsor community events.
fyi @smiklosovic , @Claudenw, @dchenbecker (spread the word)

**Comment by dchenbecker** [comment]
That sounds reasonable, @michaelsembwever . Does the linked "Must be approved by the Cassandra PMC" page have clear criteria for approval? If that link points to https://cwiki.apache.org/confluence/display/CASSANDRA/Cassandra+Project+Governance, I don't see event listing criteria there, although I think it's OK if those criteria are discussed in a separate thread.

**Comment by dchenbecker** [comment]
OK, I missed on my first reading, but it looks like it's a self-anchor to "https://www.apache.org/foundation/marks/events.html#approval", which I don't actually see in the events page adoc.

**Comment by hemidactylus** [comment]
Good point @dchenbecker ! This should be opening in a new tab, I'm going to fix that now.
As for the page this anchor links to, the page I put there is the one I used as a starting point when making sure our events were meeting all requirements before submitting them to the Apache PMC for approval.

The short text on the page ("If you are organizing ...") is there precisely to stress the goal of this page to be an aggregation point for all relevant events, with 'community' as the common denominator. In this way, a visitor can see them all (regardless of location and organizer) instead of having them scattered among blog posts.
Of course I would be more than happy to reformulate this small paragraph (where I improvised a bit) if there is preference for expressing it differently.

**Comment by dchenbecker** [comment]
@hemidactylus I like the page, but I feel like I'm still missing where in the PR you've added or linked to an existing page that lists criteria for approval of an event. As an organizer of a (hypothetical) event, it would be great to be able to review requirements before I submit something to the PMC. If we don't yet have such a page/list of criteria, I'm OK with that being a separate discussion, but I think that the current anchor target of "https://www.apache.org/foundation/marks/events.html#approval" just links back to itself, right? Sorry if I'm missing something obvious here :)

**Comment by dchenbecker** [comment]
@hemidactylus sorry, I just realized it's a separate link. That links to the Apache criteria for use of trademark, right? Is that temporary until we have something specific for the Cassandra PMC approval?

**Comment by hemidactylus** [comment]
I think at the moment this is all there is, as I am not aware of any Cassandra-specific set of requirements.
At least, the process I went through when seeking approval has been based on the general provisions listed in steps 1 through 5 on the linked paragraph.
I agree with you that a clear set of requirements and a short how-to specific to Cassandra would encourage more and better community events and make the process smoother. And probably, at that point, the Events page would be a good place to host such information. My impression, however, is that that would deserve its own PR and a separate discussion.

**Comment by hemidactylus** [comment]
> @hemidactylus sorry, I just realized it's a separate link. That links to the Apache criteria for use of trademark, right? Is that temporary until we have something specific for the Cassandra PMC approval?

Ahah, the fact that both pages are `events.html` did not help, sorry about that :)

**Comment by dchenbecker** [comment]
Agreed, we can save expanding the criteria a future discussion. Otherwise, the changes look good to me!

**Comment by hemidactylus** [comment]
I completed a redesign of the page after discussions with @ErickRamirezAU .
Now the cards are bigger and are arranged at most two per row, with a nice square image in them. Their internal spacings, etc are revised accordingly.
There is a new blurb about Cassandra Days being a community event and that contributions are welcome, from anyone, in helping run them around the world.
There is a banner on the top of the page about the Summit 2023.
The mouse pointer is back to the 'default' when hovering on the card.

This is, to me, a viable proposal. (I hereby remove the wip status from this PR).
events-B-1
events-B-2

**Review by ErickRamirezAU — APPROVED** [review]
I've suggested changes with Stefano and incorporated them in the PR. I've also squashed the commits.

File: site-content/source/modules/ROOT/pages/blog/Apache-Cassandra-4.1-Features-Guardrails-Framework.adoc
Comment (jmckenzie-dev) [inline site-content/source/modules/ROOT/pages/blog/Apache-Cassandra-4.1-Features-Guardrails-Framework.adoc:73]: Do we regularly sneak things in to PR's for website changes? For the record: definitely not against this. Just want to double check this isn't accidentally flying under the radar here and colliding with other work in flight and/or accidentally pulled over from a larger change.

File: site-content/source/modules/ROOT/pages/download.adoc
Comment (jmckenzie-dev) [inline site-content/source/modules/ROOT/pages/download.adoc:21]: How do we normally update versions on the site? Is this something we just do adhoc when we realize they're stale, or perhaps something we can integrate into our release manager pipeline and collab w/Mick / release managers on?

File: site-content/source/modules/ROOT/pages/events.adoc
Comment (jmckenzie-dev) [inline site-content/source/modules/ROOT/pages/events.adoc:30]: When we link to external non-ASF hosted sites, is there any need for a notification that you're leaving the ASF space and going to a particular vendor or other foundation's properties / sites? Genuinely asking here as I don't know and want to make sure we grease the skids on this as much as possible.

File: site-ui/src/layouts/events.hbs
Comment (dchenbecker) [inline site-ui/src/layouts/events.hbs:71]: What exactly is this trying to do that we can't just do in markup?

File: site-content/source/modules/ROOT/pages/blog/Apache-Cassandra-4.1-Features-Guardrails-Framework.adoc
Comment (hemidactylus) [reply site-content/source/modules/ROOT/pages/blog/Apache-Cassandra-4.1-Features-Guardrails-Framework.adoc:73]: This seems to be the fix to https://issues.apache.org/jira/browse/CASSANDRA-17936 which was indeed merged a few days ago already.

File: site-content/source/modules/ROOT/pages/events.adoc
Comment (hemidactylus) [reply site-content/source/modules/ROOT/pages/events.adoc:30]: Hmm, I don't know the answer. I honestly assumed that external links would be good out-of-the-box here as they are on blog posts.
(a general warning "you're leaving the site" would be best handled by client-side code, which would be a feature to implement independently from individual external links).
I took care to have external links open in new tab all the time.

File: site-ui/src/layouts/events.hbs
Comment (hemidactylus) [reply site-ui/src/layouts/events.hbs:71]: I copied this over from a similar behaviour on the blog page.
There, a search box appears on the page to filter the entries (client-side).
At the moment, this event page has not enough entries to justify a search box, but perhaps in the future it could (or a filtering based on some geography tags of items, something like that could be useful). So I did not delete it all away but kept it for future reinstatement.

Of course this can be removed completely for cleanliness, nothing against that.

File: site-content/source/modules/ROOT/pages/events.adoc
Comment (michaelsembwever) [reply site-content/source/modules/ROOT/pages/events.adoc:30]: this is ASF-approved external website. (that it's on LF does not matter.) 

otherwise, external/vendor links should be new target, and no favouritism shown.

no objections to what is done here. good and important question.

File: site-content/source/modules/ROOT/pages/download.adoc
Comment (michaelsembwever) [reply site-content/source/modules/ROOT/pages/download.adoc:21]: the release manager manually updates them
e.g. like this: https://github.com/apache/cassandra-website/commit/4212d0c8652852988d2026bb4b92c42f862a4dbf

Comment (jmckenzie-dev) [reply site-content/source/modules/ROOT/pages/download.adoc:21]: Figured as much. What are implications to these diffs in this PR here?

File: site-content/source/modules/ROOT/pages/events.adoc
Comment (jmckenzie-dev) [reply site-content/source/modules/ROOT/pages/events.adoc:30]: 👍 Makes complete sense to me. What triggered the thought was actually a completely different domain that's regulated by a ratings board etc, so quite different context. But figured I'd check.

File: site-content/source/modules/ROOT/pages/download.adoc
Comment (michaelsembwever) [reply site-content/source/modules/ROOT/pages/download.adoc:21]: I don't know why this PR is updating those dates, it shouldn't be.

Comment (ErickRamirezAU) [reply site-content/source/modules/ROOT/pages/download.adoc:21]: I honestly don't know how those snuck into this PR. They weren't part of any of the commits. I need to investigate.

File: site-ui/src/layouts/events.hbs
Comment (ErickRamirezAU) [reply site-ui/src/layouts/events.hbs:71]: @dchenbecker we were just trying to do the path of least resistance by replicating what the other pages already have because of the urgency of getting the page up. 🙂

Comment (dchenbecker) [reply site-ui/src/layouts/events.hbs:71]: > @dchenbecker we were just trying to do the path of least resistance by replicating what the other pages already have because of the urgency of getting the page up. 🙂

Fair enough :) We might consider putting this into a common doc down the road and just include it instead of copying it, but that shouldn't block this PR

File: site-content/source/modules/ROOT/pages/download.adoc
Comment (ErickRamirezAU) [reply site-content/source/modules/ROOT/pages/download.adoc:21]: I don't know what's going on but I suspect it had something to do with the `rebase` I did to squash the commits. 🤦‍♂️

Comment (ErickRamirezAU) [reply site-content/source/modules/ROOT/pages/download.adoc:21]: I gave up and submitted a new patch.

**Comment by ErickRamirezAU** [comment]
Sorry I couldn't fix @hemidactylus 's PR so I've pushed the commits to a new PR instead -- #188 .

**Pull request closed without merging**
